internal/fs/handle/file.go

```diff
@@ -316,6 +324,8 @@ func (fh *FileHandle) tryEnsureReadManager(ctx context.Context, sequentialReadSi
 	fh.destroyReadManager()
 
 	// Create a new read manager for the current inode state.
+	fh.mu.Lock()
```

Comment:
Seem we are violating the lock ordering - duplicating the comment as it's important.
Ref(lock-ordering): https://github.com/GoogleCloudPlatform/gcsfuse/blob/0c18cd143524ab58dfaee6cfb7f9fab0be05029c/internal/fs/fs.go#L329